src/Solvers/plan_transforms.jl

```diff
@@ -146,16 +146,28 @@ function plan_transforms(grid::XYZRegularRG, storage, planner_flag)
 end
 
 """ Used by FourierTridiagonalPoissonSolver. """
-function plan_transforms(grid::Union{XYRegularRG, XZRegularRG, YZRegularRG}, storage, planner_flag)
+function plan_transforms(grid, storage, planner_flag, untransformed_dim)
```

Comment:
@xkykai this is the key change. Transforms were being planned assuming something about grid regularity. This assumption is violated when we want to use a tridiagonal solver on triply uniform grid...